Src/ImageProccessing/Merge.cs

```diff
@@ -1,41 +1,73 @@
 using System;
+using System.Collections.Generic;
 using ImageMagick;
 using GpkgMerger.Src.Batching;
 using GpkgMerger.Src.Utils;
+using GpkgMerger.Src.DataTypes;
 
 namespace GpkgMerger.Src.ImageProccessing
 {
     public static class Merge
     {
-        public static string MergeNewToBase(Tile newTile, Tile baseTile)
+
+        public static string MergeTiles(List<CorrespondingTileBuilder> tiles, Coord targetCoords)
         {
-            string blob;
-            byte[] newBytes = StringUtils.StringToByteArray(newTile.Blob);
-            using (MagickImage newImage = new MagickImage(newBytes))
+            var images = new List<MagickImage>();
+            Tile lastProccessedTile = null;
+            for (var i = tiles.Count - 1; i >= 0; i--)
             {
-                if (newImage.HasAlpha)
+                MagickImage tileImage = null;
+                try
                 {
-                    byte[] baseBytes = StringUtils.StringToByteArray(baseTile.Blob);
-                    using (MagickImage baseImage = new MagickImage(baseBytes))
+                    lastProccessedTile = tiles[i]();
+                    if (lastProccessedTile.Z > targetCoords.z)
                     {
-
-                        if (baseTile.Z != newTile.Z)
-                        {
-                            Upscaling.Upscale(baseImage, baseTile, newTile);
-                        }
-
-                        baseImage.Composite(newImage, CompositeOperator.SrcOver);
-                        byte[] baseByteArr = baseImage.ToByteArray();
-                        blob = Convert.ToHexString(baseByteArr);
+                        images.ForEach(image => image.Dispose());
+                        throw new NotImplementedException("down scaling tiles is not supported");
```

Comment:
Why? Isn't this just up-scaling with opposite parameters?
In addition, this is very problematic even if it was correct. This would cause only partial merges and then Exception thrown.

Reply:
it isn't.
down scale required multiple tiles as sources so it requires different logic.
and if this condition is ever met it means we have major bag in the logic that generates the coords as targetCoords.z should be the maximum zoom of all the sources and if the tile zoom is higher it means that its zoom is higher then is zoom so there is a bug,

this is manly for safety and future prep and meant to prevent future bugs